nep-12.mediawiki

```diff
@@ -0,0 +1,2098 @@
+<pre>
+  NEP: 12
+  Title: Unified dApp API for wallet providers
+  Author: Nick Fujita <nickfujita@gmail.com>, Matus Zamborsky <zamborsky@gmail.com>
+  Type: Standard
+  Status: Accepted
+  Created: 2018-08-28
+</pre>
+
+==Abstract==
+
+This NEP describes a common API interface for dApps to communicate with external wallet providers. The use of a trusted 3rd party wallet providers will help users feel more secure when using dApps, and the unified interface will help dApp creators to have a more uniform developer experience when making their dApps compatible with various providers.
+
+
+==Motivation==
+
+===End Users===
+
+As dApps come into the ecosystem, there will be more concerns about the safety of user assets. If dApps all required users to input their private keys in order to use them, it just takes one malicious dApp to steal all their funds. By using a trusted wallet provider which interfaces with the various dApps in the ecosystem on their behalf, users can reduce the exposure of their private keys. This will even allow users to transact with their hardware wallets via the wallet provider,  never having to reveal their private keys even to the wallet itself.
+
+===dApp Developers===
+
+One of the initial hurtles for any developer when starting to develop a dApp is to create a wallet module that will allow the user and application to interface with the NEO blockchain. While there are many quality SDKs out there such as neon-js for facilitating the communication of these requests, there are often many hurtles to successfully constructing the right combination of methods, along with input and output parsing. The issue only gets amplified when trying to integrate with hardware wallet providers such as a Ledger device.
```

Comment:
hurtles -> hurdles ?
2 times in the paragraph

Reply:
(...)many quality SDKs out there such as neon-js (...) -> (...) many quality SDKs out there, such as neon-js, (...)

I am not fully sure, I usually use more commas than necessary...aehuahea

Reply:
to successfully constructing -> to successfully construct